docs/source/products/rocket-ecommerce/configuration.rst
Configuration
=============

.. title:: Rocket eCommerce - Configuration   
.. meta::
    :description: Learn how to configure Rocket eCommerce, a production-ready  
    :keywords: rocket ecommerce, stripe integration, configuration

This page explains the core settings of `Rocket eCommerce <./index.html>`__, a starter that allows to build and deploy a powerful eCommerce service without coding.  

.. include::  /_templates/components/banner-top.rst

General settings
----------------

In this section, the service ADMIN can edit:

- Site Name
- Copyright (visible in the footer)
- Hero Video - the video that plays in the hero section of the HOMEpage

Admin Settings
**************

.. image:: https://github.com/user-attachments/assets/49c7d7b9-dea6-436a-99ff-7395521bc262
   :alt: Rocket eCommerce - Admin Settings Page 

Admin Profile
**************

.. image:: https://github.com/user-attachments/assets/aef74b89-6663-4b79-9294-eb97e7aa55af
   :alt: Rocket eCommerce - Admin Profile Page

Stripe Settings
***************

.. image:: https://github.com/user-attachments/assets/864056f2-d707-40ad-93ad-3682d90bbce8
   :alt: Rocket eCommerce - Admin Settings, Edit Stripe Keys 

Import Stripe Products
**********************

.. image:: https://github.com/user-attachments/assets/a37fb794-f688-4f4e-90b5-778f88581452
   :alt: Rocket eCommerce - Import Products from Stripe  

Edit Products
*************

.. image:: https://github.com/user-attachments/assets/da2fddd5-6ee4-453c-9c6b-e595280bdcbf
   :alt: Rocket eCommerce - Settings, Edit products page

.. image:: https://github.com/user-attachments/assets/4df10d64-e525-4dbb-bb9c-dca097be101d
   :alt: Rocket eCommerce - Settings, Edit products page (more options) 

Social Links
************

.. image:: https://github.com/user-attachments/assets/108c424a-4d29-410d-adaa-4875fdee0d47
   :alt: Rocket eCommerce - Settings, Edit Social Links (visible in the footer)

.. include::  /_templates/components/footer-links.rst


docs/source/products/django/datta-able/configure-mysql.rst
Configure `MySql </docs/technologies/mysql.html>`__  
===================================================

This page explains how to change the persistence for Django Datta Able from SQLite (the default) to `MySql </docs/technologies/mysql.html>`__. 

Prerequisites
-------------

In order to switch to MySql, a server needs to be installed and running. The 2nd step is to create a database, a user and a strong password for that DB account.

All these informations are going to be used in the `.env` file. 

Manual Set up
-------------

- **Step #1**: Make sure **MySql Server** is up & running  

.. code-block:: bash  

    service mysqld status # For Linux Systems 

- **Step #2**: Create the database, and DB user 

Using the terminal or a visual tool like **phpmyadmin** or `DBeaver <https://dbeaver.io/>`__ create the following:

.. code-block:: text  

    - a new database 
    - a new user 
    - a strong password 
    - give full privileges for the DB user over the new database 

- **Step #3**: Update the `.env` file located in the root of Django Datta Source code as suggested below 

.. code-block:: text 

    # True for development, False for production
    DEBUG=True

    SECRET_KEY=<STRONG_KEY_HERE>

    DB_ENGINE=mysql
    DB_HOST=localhost
    DB_NAME=YOUR_DB_HERE
    DB_USERNAME=YOUR_DB_USER_HERE
    DB_PASS=YOUR_DB_PASS_HERE
    DB_PORT=3306    

In case the service runs on another port and not on 3306 (default one), don't forget to adjust. 

- **Step #4** - Install the MySql Python Driver 

.. code-block:: bash 

    pip install mysqlclient


- **Step #5** - Migrate the DataBase 

.. code-block:: bash 

    python manage.py makemigrations     # Migrate DataBase (generate tables) 
    python manage.py migrate            # Apply Changes on Database 

If the above steps are executed without errors, the MySql database should contain the tables 

- **Step #6** - Start the project using MySql 

.. code-block:: bash 

    python manage.py runserver

At this point, the app runs at **http://127.0.0.1:8000/** using MySql as DB Server 

.. include::  /_templates/components/generator-django.rst
    
.. include::  /_templates/components/footer-links.rst


CHANGELOG.md
# Change Log

## [0.0.82] 2025-03-03
### [App Generator](https://app-generator.dev/) Changes

- New Products
  - [Rocket eCommerce](https://app-generator.dev/product/rocket-ecommerce/django/)

## [0.0.81] 2025-03-03
### [App Generator](https://app-generator.dev/) Changes

- [Flask Generator](https://app-generator.dev/tools/flask-generator/)
  - Update page & PRO popup
- [Django Generator](https://app-generator.dev/tools/django-generator/)
  - Update page & PRO popup

## [0.0.80] 2025-03-03
### [App Generator](https://app-generator.dev/) Changes

- [Flask Generator](https://app-generator.dev/tools/flask-generator/) - new service
- [Django Generator](https://app-generator.dev/tools/django-generator/) - update the access

## [0.0.79] 2025-02-26
### [App Generator](https://app-generator.dev/) Changes

- Added CLI Flask Generator, options
  - Edit Database
  - Extended User Model
  - Docker, CI/CD
  - Dynamic Tables
  - OAuth: Google & GitHub

## [0.0.78] 2025-02-23
### [App Generator](https://app-generator.dev/) Changes

> Documentation Update - Programming Concepts

- [Programming Concept - Metaprogramming](https://app-generator.dev/docs/concepts/metaprogramming.html)
- [Programming Concept - Reactive Programming](https://app-generator.dev/docs/concepts/reactive-programming.html)
- [Programming Concept - N-Tier Architecture](https://app-generator.dev/docs/concepts/n-tier-architecture.html)
- [Programming Concept - Microservice](https://app-generator.dev/docs/concepts/microservices.html)
- [Programming Concept - Functional Programming](https://app-generator.dev/docs/concepts/functional-programming.html)

## [0.0.77] 2025-02-23
### [App Generator](https://app-generator.dev/) Changes

- [Vite Documentation](https://app-generator.dev/docs/technologies/vite.html)
  - [Getting Started](https://app-generator.dev/docs/technologies/vite/index.html)
  - [Vite vs. Webpack](https://app-generator.dev/docs/technologies/vite/vite-vs-webpack.html)
- [DaisyUI Documentation](https://app-generator.dev/docs/technologies/daisyui.html)
  - [Getting Started](https://app-generator.dev/docs/technologies/daisyui/index.html)
  - [DaisyUI vs. Flowbite](https://app-generator.dev/docs/technologies/daisyui/daisyui-vs-flowbite.html)
- [Django Documentation](https://app-generator.dev/docs/technologies/django.html)
  - [Integrate Django, DaisyUI, and Vite](https://app-generator.dev/docs/technologies/django/integrate-daisyui.html) 

## [0.0.76] 2025-02-21
### [App Generator](https://app-generator.dev/) Changes

- [Flask Datta Able](https://app-generator.dev/product/datta-able/flask/) Update
  - Added [Dynamic DataTable](https://flask-datta-demo.onrender.com/dynamic-dt) Module
  - Update [Documentation](https://app-generator.dev/docs/products/flask/datta-able/index.html)

## [0.0.75] 2025-01-28
### [App Generator](https://app-generator.dev/) Changes

- Update [Custom Development](https://app-generator.dev/custom-development/) page

## [0.0.74] 2025-01-27
### [App Generator](https://app-generator.dev/) Changes

- Update [Custom Development](https://app-generator.dev/custom-development/) page

## [0.0.73] 2025-01-20
### [App Generator](https://app-generator.dev/) Changes

- Celery Optimizations
  - Added Concurrency
  - Added Gevent for Parallelism
  - Disable Beat Schedule (incompatible with Gevent Parallelism)

## [0.0.72] 2025-01-19
### [App Generator](https://app-generator.dev/) Changes

- Docker Optimizations
- Increase Celery TASKS Timeout: 90s

## [0.0.71] 2025-01-16
### [App Generator](https://app-generator.dev/) Changes

- Fix [CSV Processor](https://app-generator.dev/tools/csv-processor/)

## [0.0.70] 2025-01-14
### [App Generator](https://app-generator.dev/) Changes

- Added Files Manager [ under the settings menu ]
  - File Types: `json`, `yaml`, `csv`, `xls`, `doc`, `docx`   

## [0.0.69] 2025-01-14
### [App Generator](https://app-generator.dev/) Changes

- [DB Migrator](https://app-generator.dev/tools/db-migrator/) - Stability Fixes
- [DB Processor](https://app-generator.dev/tools/db-processor/) - Stability Fixes

## [0.0.68] 2025-01-13
### [App Generator](https://app-generator.dev/) Changes

- Fix The Issue System
- Update `anthropic` version: `v0.42.0`
- Added API Generator Endpoint

## [0.0.67] 2025-01-08
### [App Generator](https://app-generator.dev/) Changes

- Update [Terms & Pricing](https://app-generator.dev/terms/)
- **Pro Subscription** - $4.99/mo
  - Unlock a New Premium Starter at each 4mo
  - [Onboarding Kit for Developers](https://app-generator.dev/onboarding-kit/)
  - Premium Support
  - Full access to the [dev tools](https://app-generator.dev/tools/django-generator/) 
- **Company Subscription** - $39.99/mo
  - All included in `PRO Subscription`, plus
  - Deployment (CI/CD) for AWS, Azure, DO, and Render
  - On-premise Deployment (your server)
  - All packages include CI/CD Scripts for GitHub, BitBucket   

## [0.0.66] 2025-01-08
### [App Generator](https://app-generator.dev/) Changes

- [CSV Processor](https://app-generator.dev/tools/csv-processor/) 
  - Fixes and added Limits
- [Django App Generator](https://app-generator.dev/tools/django-generator/)
  - Allow unlimited apps for Guests 

## [0.0.65] 2025-01-06
### [App Generator](https://app-generator.dev/) Changes

- [Django App Generator](https://app-generator.dev/tools/django-generator/)
  - Improve Error Checking
  - Code refactoring 

## [0.0.64] 2025-01-04
### [App Generator](https://app-generator.dev/) Changes

- Update [Service Terms & Pricing](https://app-generator.dev/terms/)
- **Pro Subscription** - $4.99/mo
  - Unlock a New Premium Starter at each 4mo
  - Added [Onboarding Kit for Developers](https://app-generator.dev/onboarding-kit/)
  - Premium Support
  - Full access to the [dev tools](https://app-generator.dev/tools/django-generator/) 

## [0.0.63] 2024-12-16

core/settings.py
"""
Django settings for core project.

Generated by 'django-admin startproject' using Django 4.1.2.

For more information on this file, see
https://docs.djangoproject.com/en/4.1/topics/settings/

For the full list of settings and their values, see
https://docs.djangoproject.com/en/4.1/ref/settings/
"""

import os, random, string, json
from pathlib import Path
from django.contrib import messages
from str2bool import str2bool 
from dotenv import load_dotenv

load_dotenv()  # take environment variables from .env.

# Build paths inside the project like this: BASE_DIR / 'subdir'.
BASE_DIR = Path(__file__).resolve().parent.parent

# Quick-start development settings - unsuitable for production
# See https://docs.djangoproject.com/en/4.1/howto/deployment/checklist/

# SECURITY WARNING: keep the secret key used in production secret!
SECRET_KEY = os.environ.get('SECRET_KEY')
if not SECRET_KEY:
    SECRET_KEY = '426967205465636820436F6D70616E69657320616E6420564373204B696C6C20476F6F64204C6966650A'

PRO_SUBSCRIPTION_PRICE = os.environ.get('PRO_SUBSCRIPTION_PRICE', '4.99')
PRO_SUBSCRIPTION_COMPANY_PRICE = os.environ.get('PRO_SUBSCRIPTION_COMPANY_PRICE', '39.99')
CUST_DEV_WEEK_PRICE = os.environ.get('CUST_DEV_WEEK_PRICE', '899')
CUST_DEV_HOUR_PRICE = os.environ.get('CUST_DEV_HOUR_PRICE', '29')
ONBOARDING_KIT_PRICE = os.environ.get('ONBOARDING_KIT_PRICE', '19.99')

# Enable/Disable DEBUG Mode
DEBUG = str2bool(os.environ.get('DEBUG'))

# Hosts Settings
ALLOWED_HOSTS = [ 'localhost', '127.0.0.1', '81.181.87.190', '0.0.0.0', 'app-generator.dev', 'development.app-generator.dev' ]
CSRF_TRUSTED_ORIGINS = [ 'http://localhost:8000', 'http://localhost:5085', 'http://127.0.0.1:8000', 'http://127.0.0.1:5085', 'https://app-generator.dev', 'http://app-generator.dev', 'http://development.app-generator.dev', 'https://development.app-generator.dev']

# Application definition

INSTALLED_APPS = [
    "webpack_loader",
    "frontend",
    
    "django.contrib.admin",
    "django.contrib.auth",
    "django.contrib.contenttypes",
    "django.contrib.sessions",
    "django.contrib.messages",
    "django.contrib.staticfiles",

    # CLI
    "cli",
    "rest_framework",
    "rest_framework.authtoken",
    "drf_yasg",
    "django_filters",
    "corsheaders",

    # APPS
    "apps.ai_processor",
    "apps.api",
    "apps.common",
    "apps.pages",
    "apps.authentication",
    "apps.blog",
    "apps.dashboard",

    "apps.products",
    "apps.tasks",
    "apps.support",

    "apps.tools",
    "apps.tool_db_migrator",
    "apps.tool_django_generator",
    "apps.tool_flask_generator",

    # Util
    "debug_toolbar",
    "docs",
    "sslserver",

    # allauth apps
    'allauth',
    'allauth.account',
    'allauth.socialaccount',
    'allauth.socialaccount.providers.github',
    'allauth.socialaccount.providers.google',

    'django_quill',
    'django_celery_results',
    'django.contrib.sitemaps',
]

MIDDLEWARE = [
    "django.middleware.security.SecurityMiddleware",
    "whitenoise.middleware.WhiteNoiseMiddleware",
    "django.contrib.sessions.middleware.SessionMiddleware",
    "corsheaders.middleware.CorsMiddleware",
    "django.middleware.common.CommonMiddleware",
    "django.middleware.csrf.CsrfViewMiddleware",
    "django.contrib.auth.middleware.AuthenticationMiddleware",
    "django.contrib.messages.middleware.MessageMiddleware",
    "django.middleware.clickjacking.XFrameOptionsMiddleware",

    # Util
    "debug_toolbar.middleware.DebugToolbarMiddleware",

    # allauth 
    "allauth.account.middleware.AccountMiddleware",

    # Custom
    "apps.dashboard.log_middleware.APILoggingMiddleware",
]

CORS_ALLOW_ALL_ORIGINS = True

ROOT_URLCONF = "core.urls"

UI_TEMPLATES = os.path.join(BASE_DIR, 'templates') 

TEMPLATES = [
    {
        "BACKEND": "django.template.backends.django.DjangoTemplates",
        "DIRS": [UI_TEMPLATES],
        "APP_DIRS": True,
        "OPTIONS": {
            "context_processors": [
                "django.template.context_processors.debug",
                "django.template.context_processors.request",
                "django.contrib.auth.context_processors.auth",
                "django.contrib.messages.context_processors.messages",

                "apps.common.context_processors.profile_context",
                "apps.common.context_processors.version_context",
                "apps.common.context_processors.props_context",
                "apps.common.context_processors.google_tag",

                "apps.common.context_processors.price_subscription_pro",
                "apps.common.context_processors.price_subscription_company",
                "apps.common.context_processors.price_cust_dev_week",
                "apps.common.context_processors.price_cust_dev_hour",
                "apps.common.context_processors.price_onboarding_kit",
            ],
        },
    },
]

WSGI_APPLICATION = "core.wsgi.application"


# Database
# https://docs.djangoproject.com/en/4.1/ref/settings/#databases

DB_ENGINE   = os.environ.get('DB_ENGINE')
DB_USERNAME = os.environ.get('DB_USERNAME')
DB_PASS     = os.environ.get('DB_PASS')
DB_HOST     = os.environ.get('DB_HOST')
DB_PORT     = os.environ.get('DB_PORT')
DB_NAME     = os.environ.get('DB_NAME')

if DB_ENGINE and DB_NAME and DB_USERNAME:
    DATABASES = { 
        'default': {
            'ENGINE'  : 'django.db.backends.' + DB_ENGINE, 
            'NAME'    : DB_NAME,
            'USER'    : DB_USERNAME,
            'PASSWORD': DB_PASS,
            'HOST'    : DB_HOST,
            'PORT'    : DB_PORT,
        },          
    }
else:
    DATABASES = {
        'default': {
            'ENGINE': 'django.db.backends.sqlite3',
            'NAME'  : 'db.sqlite3',
        },
        # 'db_src': {
        #     'ENGINE'  : 'django.db.backends.'  + os.environ.get('DB_SRC_ENGINE', 'NOT_SET'), 
        #     'NAME'    : os.environ.get('DB_SRC_NAME' , None),
        #     'USER'    : os.environ.get('DB_SRC_USERNAME', None),
        #     'PASSWORD': os.environ.get('DB_SRC_PASS', None),
        #     'HOST'    : os.environ.get('DB_SRC_HOST', None),
        #     'PORT'    : os.environ.get('DB_SRC_PORT', None),
        # },
        # 'db_dest': {
        #     'ENGINE'  : 'django.db.backends.'  + os.environ.get('DB_DEST_ENGINE', 'NOT_SET'), 
        #     'NAME'    : os.environ.get('DB_DEST_NAME' , None),
        #     'USER'    : os.environ.get('DB_DEST_USERNAME', None),
        #     'PASSWORD': os.environ.get('DB_DEST_PASS', None),
        #     'HOST'    : os.environ.get('DB_DEST_HOST', None),
        #     'PORT'    : os.environ.get('DB_DEST_PORT', None),
        # },
    } 


# Password validation
# https://docs.djangoproject.com/en/4.1/ref/settings/#auth-password-validators

AUTH_PASSWORD_VALIDATORS = [
    {
        "NAME": "django.contrib.auth.password_validation.UserAttributeSimilarityValidator",
    },
    {
        "NAME": "django.contrib.auth.password_validation.MinimumLengthValidator",
    },
    {
        "NAME": "django.contrib.auth.password_validation.CommonPasswordValidator",
    },
    {
        "NAME": "django.contrib.auth.password_validation.NumericPasswordValidator",
    },
]


# Internationalization
# https://docs.djangoproject.com/en/4.1/topics/i18n/

LANGUAGE_CODE = "en-us"

TIME_ZONE = "UTC"

USE_I18N = True

USE_TZ = True


# Static files (CSS, JavaScript, Images)
# https://docs.djangoproject.com/en/4.1/howto/static-files/

STATIC_URL = "static/"

STATIC_ROOT = os.path.join(BASE_DIR, 'staticfiles')

STATICFILES_DIRS = (
    os.path.join(BASE_DIR, 'static'),
)

MEDIA_URL = 'media/'
MEDIA_ROOT = os.path.join(BASE_DIR, 'media')

# Default primary key field type
# https://docs.djangoproject.com/en/4.1/ref/settings/#default-auto-field

DEFAULT_AUTO_FIELD = "django.db.models.BigAutoField"

MESSAGE_TAGS = {
    messages.INFO: 'text-blue-800 border border-blue-300 bg-blue-50 dark:text-blue-400 dark:border-blue-800',
    messages.SUCCESS: 'text-green-800 border border-green-300 bg-green-50 dark:text-green-400 dark:border-green-800',
    messages.WARNING: 'text-yellow-800 border border-yellow-300 bg-yellow-50 dark:text-yellow-300 dark:border-yellow-800',
    messages.ERROR: 'text-red-800 border border-red-300 bg-red-50 dark:text-red-400 dark:border-red-800',
}

SITE_ID = 1
LOGIN_REDIRECT_URL = "/dashboard/profile/"

# OAuth
GITHUB_CLIENT_ID   = os.getenv("GITHUB_CLIENT_ID" , "")
GITHUB_SECRET_KEY  = os.getenv("GITHUB_SECRET_KEY", "") 

# Upload 
GITHUB_API_KEY     = os.getenv("GITHUB_API_KEY", None) 
GITHUB_API_ACCOUNT = os.getenv("GITHUB_API_ACCOUNT", 'https://github.com/appseed-projects4/') 

SOCIALACCOUNT_PROVIDERS = {
    'github': {
        'SCOPE': [
            'read:user',
            'user:email'
        ],        
        "APP": {
            "client_id": GITHUB_CLIENT_ID,
            "secret": GITHUB_SECRET_KEY,
        },
    },
}

SOCIALACCOUNT_QUERY_EMAIL = True

QUILL_CONFIGS = {
    'default':{
      "theme": "snow",
      "modules": {
        "syntax": True,
        "toolbar": [
          [
            {"header": []},
            {"align": []},
            "bold",
            "italic",
            "underline",
            "strike",
            "blockquote",
          ],
          ["code-block", "link"],
        ],
        "imageCompressor": {
          "quality": 0.8,
          "maxWidth": 2000,
          "maxHeight": 2000,
          "imageType": "image/jpeg",
          "debug": False,
          "suppressErrorLogging": True
        },
        "resize": {
          "showSize": True,
          "locale": {}
        }
      },
      "formats": [
        "header",
        "bold",
        "italic",
        "underline",
        "strike",
        "blockquote",
        "code-block",
        "link",
        "indent",
        "list",
        "align",
      ],
      "sanitize": True,
    }
}

# ### Async Tasks (Celery) Settings ###

CELERY_SCRIPTS_DIR        = os.path.join(BASE_DIR, "tasks_scripts" )

# CELERY_LOGS_URL           = "/tasks_logs/"
# CELERY_LOGS_DIR           = os.path.join(BASE_DIR, "tasks_logs"    )

CELERY_BROKER_URL         = os.environ.get("CELERY_BROKER", "redis://127.0.0.1:6379/0")
CELERY_RESULT_BACKEND     = os.environ.get("CELERY_BROKER", "redis://127.0.0.1:6379/0")

CELERY_TASK_TRACK_STARTED = True
CELERY_TASK_TIME_LIMIT    = 30 * 60
CELERY_CACHE_BACKEND      = "django-cache"
CELERY_RESULT_BACKEND     = "django-db"
CELERY_RESULT_EXTENDED    = True
CELERY_RESULT_EXPIRES     = 60*60*24*30 # Results expire after 1 month
CELERY_ACCEPT_CONTENT     = ["json"]
CELERY_TASK_SERIALIZER    = 'json'
CELERY_RESULT_SERIALIZER  = 'json'
CELERY_BROKER_CONNECTION_RETRY_ON_STARTUP = True
########################################


# DOCS_ROOT = os.path.join(BASE_DIR, 'docs/build/html')

DOCS_ROOT = os.path.join(BASE_DIR, 'docs/build/html')
DOCS_ACCESS = 'public'

#EMAIL_BACKEND = 'django.core.mail.backends.console.EmailBackend'
EMAIL_BACKEND = 'django.core.mail.backends.smtp.EmailBackend'
EMAIL_HOST=os.environ.get('EMAIL_HOST')
EMAIL_PORT=465
EMAIL_USE_TLS = False
EMAIL_USE_SSL = True
EMAIL_HOST_USER=os.environ.get('EMAIL_HOST_USER')
EMAIL_HOST_PASSWORD=os.environ.get('EMAIL_HOST_PASSWORD')
EMAIL_DISPLAY_NAME=os.environ.get('EMAIL_DISPLAY_NAME')

REST_FRAMEWORK = {
    'DEFAULT_PERMISSION_CLASSES': [
        'rest_framework.permissions.AllowAny'
    ],
    'DEFAULT_AUTHENTICATION_CLASSES': [
        'rest_framework.authentication.TokenAuthentication',
    ],
    #'DEFAULT_THROTTLE_CLASSES': [
    #    'apps.api.throttling.OncePerMinuteThrottle',
    #],
    #'DEFAULT_THROTTLE_RATES': {
    #    'once_per_minute': '1/min',
    #},
    'DEFAULT_PAGINATION_CLASS': 'rest_framework.pagination.LimitOffsetPagination',
    'PAGE_SIZE': 20,
}

GUMROAD_ACCESS_TOKEN = os.environ.get('GUMROAD_ACCESS_TOKEN')
ANTHROPIC_API_KEY = os.environ.get('ANTHROPIC_API_KEY')

WEBPACK_LOADER = {
    "DEFAULT": {
        "BUNDLE_DIR_NAME": "frontend/",
        "STATS_FILE": os.path.join(BASE_DIR, "webpack-stats.json"),
    }
} 

VERSION = open('VERSION', 'r').readline()

LIMIT_AI_PROMPT_GUEST = os.environ.get('LIMIT_AI_PROMPT_GUEST', 3)
LIMIT_AI_PROMPT_AUTH  = os.environ.get('LIMIT_AI_PROMPT_AUTH' , 5)

# Redirects file
REDIRECTS_PATH = os.path.join('templates', 'redirects.json')
REDIRECTS = json.loads( open( REDIRECTS_PATH, 'r').read() )

# Limits
CSV_PROCESS_LIMIT = os.environ.get('CSV_PROCESS_LIMIT', 20)
X_FRAME_OPTIONS = 'SAMEORIGIN'

# 
GOOGLE_TAG = os.environ.get('GOOGLE_TAG', None)